File: 0771-jewels-and-stones/0771-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        count = 0
        allJewels = set()

        for i in range(len(jewels)):
            allJewels.add(jewels[i])

        for i in range(len(stones)):
            if stones[i] in allJewels:
                count += 1

        return count

File: 0771-jewels-and-stones/0771-jewels-and-stones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
    let count = 0;
    const set = new Set();

    for (let i = 0; i < jewels.length; i++) {
        set.add(jewels[i]);
    }

    for (let i = 0; i < stones.length; i++) {
        if (set.has(stones[i])) {
            count++;
        }
    }

    return count;
};